src/operator/audit.ts

```diff
@@ -86,6 +86,9 @@ class AuditSubscriber<T, R> extends OuterSubscriber<T, R> {
         this.destination.error(errorObject.e);
       } else {
         this.add(this.throttled = subscribeToResult(this, duration));
+        if (this.throttled.closed) {
```

Comment:
When I ran this code under a debugger, that's exactly what happened: inside `subscribeToResult`, the observable completed, so by the time it is returned, it is already completed, so we shouldn't throttle subsequent results anymore.

I have two questions:
1. How should we solve this problem?
2. How do we avoid this problem elsewhere as the cold('|') observable used in tests doesn't complete immediately.